proposals/0000-android-12-issues.md

```diff
@@ -0,0 +1,277 @@
+
+### Bluetooth Runtime Permissions
+
+The bluetooth logic in the library will now require these 2 new runtime permissions. With these permissions granted there is not any issue with the bluetooth logic but we now need to account for the use cases where a user has denied permissions. Specifically without these permissions the `SdlBroadcastReceiver` will never receive Intents with the `ACL_CONNECTED` action and the `SdlRouterService` will not be able to start the `bluetoothTransport`.
+
+### AndroidManifest Exported Flag
+
+These attributes were already defined where required. 
+
+### PendingIntent Mutable Flag
+
+The `SdlRouterService` will crash if the the app targets Android 12 and does not have this flag defined on the PendingIntent used for `SdlRouterService` notifications.
+
+### Service Notification Delays
+
+These notifications may be delayed by Android by up to 10 seconds
+
+## Alternatives considered
+
+### Foreground Services
+
+Alternatives for the Foreground Service restrictions are limited. We either need to start the `SdlService` from a foreground context or the conditions need to meet one of the exceptions listed by Google. These conditions include:
+
+* Starting the `SdlService` from an Activity.
+* Starting the `SdlService` from user interaction with a notification
+* Requesting the user ignores battery optimizations for each SDL Application
+
+These options would either require an Activity to be launched for each SDL app, the user to interact with a notification for each SDL app, or for the user to choose battery optimization options for each app which then creates a situation where the user dictates if the given app will have its `SdlService` start when the `SdlRouterService` connects.
+
+If we decide to use `PendingIntents` and export the `SdlService` we could also create a custom SDL `<permission>` to be used by apps as a requirement to be able to start the `SdlService` but nothing is stopping developers from listing that custom permission in their `AndroidManifest.xml`.   
+
+### Bluetooth Runtime Permissions
+
+In the use case of the device only being connected over USB we could try to start the `SdlRouterService` of an application that does have Bluetooth Permissions to ensure we select a `SdlRouterService` that can start the `bluetoothTransport`. The problem with this solution is when we choose to start a `SdlRouterService` that is different than the one started by the `USBAccessoryAttachmentActivity`, the `USBTransferProvider` will try to bind to the wrong `RouterService`. This will require a larger refactor of the `USBTransferProvider`.
+
+### PendingIntent Mutable Flag
+
+The one case where PendingIntents were being used previously in the library we can set the flag to be `PendingIntent.FLAG_MUTABLE` but as we are handing the PendingIntent off to the notificationManager and the intent does not need to be changed at any point we should use `PendingIntent.FLAG_IMMUTABLE`.
```

Comment:
I am thinking `FLAG_IMMUTABLE` is a clear winner here so I suggest removing the section